diff --git a/src/apps/external_executor/protobuf/CMakeLists.txt b/src/apps/external_executor/protobuf/CMakeLists.txt
index d35fe3537..3418d7be5 100644
--- a/src/apps/external_executor/protobuf/CMakeLists.txt
+++ b/src/apps/external_executor/protobuf/CMakeLists.txt
@@ -38,24 +38,6 @@ foreach(proto_file ${PROTO_FILES})
             ${CMAKE_CURRENT_BINARY_DIR}/${PROTO_NAME_WE}.pb.cc
   )

-  add_custom_command(
-    OUTPUT
-      ${CMAKE_SOURCE_DIR}/tests/external_executor/${PROTO_NAME_WE}_pb2.py
-      ${CMAKE_SOURCE_DIR}/tests/external_executor/${PROTO_NAME_WE}_pb2_grpc.py
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/build.sh ${proto_file}
-            ${CMAKE_SOURCE_DIR}/tests/external_executor/
-    COMMENT "Generate Python source file from protobuf file ${PROTO_NAME}"
-    DEPENDS ${proto_file}
-    JOB_POOL one_job
-  )
-  add_custom_target(
-    ${PROTO_NAME_WE}_proto_python ALL
-    DEPENDS
-      ${CMAKE_SOURCE_DIR}/tests/external_executor/${PROTO_NAME_WE}_pb2.py
-      ${CMAKE_SOURCE_DIR}/tests/external_executor/${PROTO_NAME_WE}_pb2_grpc.py
-    JOB_POOL one_job
-  )
-
   # For now, copy .proto files across to external executor test
   configure_file(
     ${proto_file}
