diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b2a143ae..4a0cdcec9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1052,6 +1052,3 @@ install(FILES ${CMAKE_BINARY_DIR}/cmake/${CCF_PROJECT}-config.cmake
               ${CMAKE_BINARY_DIR}/cmake/${CCF_PROJECT}-config-version.cmake
         DESTINATION ${CMAKE_INSTALL_PREFIX}/cmake
 )
-
-# Perf tool executable
-include(${CCF_DIR}/tests/perf-system/submitter/CMakeLists.txt)
